src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -135,6 +137,45 @@ public boolean isControlled(UUID workspaceId, UUID referenceId) {
     }
   }
 
+  public boolean updateDataReference(
+      UUID workspaceId, UUID referenceId, String name, String referenceDescription) {
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("id", referenceId.toString())
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("name", name)
+            .addValue("reference_description", referenceDescription);
+
+    StringJoiner updateStatement =
+        new StringJoiner(
+            ", ",
+            "UPDATE workspace_data_reference SET ",
+            " WHERE reference_id = :id AND workspace_id = :workspace_id");
+
+    if (name != null) {
+      updateStatement.add("name = :name");
+    }
+    if (referenceDescription != null) {
+      updateStatement.add("reference_description = :reference_description");
+    }
```

Comment:
if name and referenceDescription are both null, this should not go to the database at all ... especially since it will generate invalid SQL `UPDATE workspace_data_reference SET  WHERE reference_id = :id AND workspace_id = :workspace_id`

Reply:
I'm going to add a guard for that at the controller level.